Added content to Init() and Update()

Kortigus opened the pull request:
Added initial content to Init() and Update()

Kortigus committed:
Added content to Init() and Update()

ExtremeSkepticist commented on NumberBaseball/Source.c:
```diff
@@ -0,0 +1 @@
+#include <stdio.h>
```
Delete this file

Kortigus replied:
Why get rid of stdio.h? if i don't have that then i can't use sscanf_s right?
edit: Nevermind, you meant delete source.c

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+	int random1;
+	int random2;
+	int random3;
+	int *n1 = &random1;
```
Do you need these pointers to pass the addresses? Why don't you just pass addresses instead of declaring pointers

Kortigus replied:
Yeah that makes more sense, thanks

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		/* Render(); */
 	}
-	Release();
+	/* Release(); */
```
Don't comment out even if there's no code in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
-		Update();
-		Render();
+		Update(n1, n2, n3);
+		/* Render(); */
```
Dont comment out even if there's no code in there

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+	}
+}
+
+/*
```
Don't comment out

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+*/
+
+/*
+void Release() {
```
Don't comment out too

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
-void Init()
+#define LINESIZE (1024)
+
+void Init(int *number1, int *number2, int *number3)
```
Follow the coding standard when declaring pointers

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		*number3 = test3;
+		break;
+	}
+	printf("Welcome to Number Baseball!\n");
```
Everything that you have to render should be done in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+	int test3;
+	while (1)
+	{
+		test1 = rand() % 10;
```
I don't think rand() will always return random values...look up srand()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
 	{
-		Update();
-		Render();
+		Update(n1, n2, n3);
```
Same as Init. Do you need to pass using pointer variables?

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		{
+			int correctCount = 0;
+
+			if (guess1 == *number1)
```
You can make functions for things below. Don't have to put all in Update()

ExtremeSkepticist replied:
Too much if statements...you can simplify

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			{
+				printf("Try again!\n");
+			}
+			clearerr(stdin);
```
Do you need to invoke this?

Kortigus replied:
guess not

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+	printf("Welcome to Number Baseball!\n");
+}
+
+void Update(int *number1, int *number2, int *number3) {
```
Same as Init(). Follow the coding standard.

ExtremeSkepticist replied:
Indent

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			}
+			clearerr(stdin);
+		}
+		else if (numbersEntered > 3 || numbersEntered < 3)
```
Do you need to check numberEntered is greater than 3 or less than 3?

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		}
+		else if (numbersEntered > 3 || numbersEntered < 3)
+		{
+			printf("Invalid amount of numbers. Please choose 3 numbers.\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			}
+			else if (guess3 == *number1 || guess3 == *number2)
+			{
+				printf("Out!\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			}
+			else
+			{
+				printf("Ball!\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			}
+
+			if (correctCount == 3) {
+				printf("You win!\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+
+			if (correctCount == 3) {
+				printf("You win!\n");
+				exit(1);
```
Use return

Kortigus replied:
But if i use return, that just brings me out to the while loop that starts updating again. Should i move the contents of Init() into Update() to keep playing the game by picking new numbers for the computer? And if so, doesn't that leave Init() kinda empty? what exactly goes in Init()?

ExtremeSkepticist replied:
1. Init() just only initializes data for the game.
2.If it keeps updating, why dont u try modifying while loop condition in main?

Kortigus replied:
so in this case should i use malloc (etc) or is this project too simple to merit that

ExtremeSkepticist replied:
> so in this case should i use malloc (etc) or is this project too simple to merit that

You don't even need to use malloc for the project

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			}
+			else
+			{
+				printf("Try again!\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		guess1 = -1;
+		guess2 = -1;
+		guess3 = -1;
+		printf("Choose 3 numbers:\n");
```
Should be in Render()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+				printf("Strike! ");
+				correctCount++;
+			}
+			else if (guess1 == *number2 || guess1 == *number3)
```
This is a Ball.

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+				printf("Strike! ");
+				correctCount++;
+			}
+			else if (guess2 == *number1 || guess2 == *number3)
```
Same as the if statement above

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+				printf("Strike!\n");
+				correctCount++;
+			}
+			else if (guess3 == *number1 || guess3 == *number2)
```
Same as the if statement above

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+	int guess1;
+	int guess2;
+	int guess3;
+	while (1) {
```
Indent

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+
+void Init(int *number1, int *number2, int *number3)
 {
+	int test1;
```
Use unsigned type for the integer variables

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		test2 = rand() % 10;
+		test3 = rand() % 10;
+
+		if (test1 == test2 || test1 == test3 || test2 == test3)
```
Users would be happy if you print out some message that they can't type same numbers :)

Kortigus replied:
this is making sure the random generator doesn't pick 3 numbers that are the same, not user input

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
 
 int main(void)
 {
+	int random1;
```
Use unsigned type. We are not storing negative numbers

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		if ((numbersEntered = sscanf(line, "%d %d %d",
+			&guess1, &guess2, &guess3)) == 3)
+		{
+			int correctCount = 0;
```
Instead of this, can you show users how many strikes & balls& outs they've got?
like
2 strikes, 1 ball, 0 out

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+}
+
+void Update(int *number1, int *number2, int *number3) {
+	int guess1;
```
Use unsigned type for the variables

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+		guess2 = -1;
+		guess3 = -1;
+		printf("Choose 3 numbers:\n");
+		if (!fgets(line, LINESIZE, stdin)) {
```
Indent

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+			printf("\n");
+			continue;
+		}
+		if ((numbersEntered = sscanf(line, "%d %d %d",
```
Do you need numbersEntered?

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,20 +1,141 @@
+				printf("Ball!\n");
+			}
+
+			if (correctCount == 3) {
```
Indent

Kortigus committed:
-Moved over content from Update() to Render()
-Edited content in Update()

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -1,13 +1,17 @@
 	int test1;
 	int test2;
 	int test3;
+	time_t t;
```
Use srand((unsigned)time(NULL));

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
-	int *n1 = &random1;
-	int *n2 = &random2;
-	int *n3 = &random3;
+	int guess1;
```
What about just grouping related variables. LIke playerScores, computerNums, playerNums..

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
+	else
+	{
+		int correctCount = 0;
+		int* guess[] = {guess1, guess2, guess3};
```
Why don't you just make arrays in main and then pass it to functions?
Function parameters are too many

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
 			clearerr(stdin);
 			printf("\n");
-			continue;
+			exit(1);
```
Users should type again if they typed the invalid

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
+		else
 		{
-			int correctCount = 0;
+			*entry = -1;
```
You don't need else statement

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
 				correctCount++;
 			}
-			else if (guess2 == *number1 || guess2 == *number3)
+			else if (*guess[i] == *number[0] || *guess[i] == *number[1] || *guess[i] == *number[2])
```
I don't think you understand the rule well.
If one of player's numbers and one of computer's numbers are same but the order each is different, 
it's a ball.

For example,
computer: 1 3 5
player:       3 9 0
Player gets one ball and two outs because they both have 3 but the orders are different.
the rest of numbers each don't match with any number of computer's, Player gets two outs

What you coded in else if statement is for checking Ball

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -24,118 +28,111 @@ void Init(int *number1, int *number2, int *number3)
+			return 1;
 		}
-		else if (numbersEntered > 3 || numbersEntered < 3)
+		else
```
You don't need else here

Kortigus committed:
Updated Init(), Update(), and Render(), so that they pass in structs instead of a huger list of pointers

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -2,15 +2,25 @@
+void Init(guess* player, guess* computer)
+{
+
+	size_t test1;
```
You don't need test1,test2,test3. Just initialize members of computer directly

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
-
-		if (!fgets(line, LINESIZE, stdin))
+		char line[LINE_SIZE];
+		size_t test1 = -1;
```
There's no point to have these variables(test1, test2, test3, guess1, guess2, guess3).
Just use player structure pointer

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
 	else if (*entry == -1)
 	{
-		printf("Invalid amount of numbers. Please choose 3 numbers.\n");
+		printf("Invalid amount of numbers.\n Please choose 3 unique numbers from 0 to 9.\n");
```
Where do you validate if all user inputs are between 0 and 9?

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
+		size_t strikeCount = 0;
+		size_t outCount = 0;
+		size_t ballCount = 0;
+		size_t* guess[] = {player->guess1, player->guess2, player->guess3};
```
This is wrong..you are storing size_t variables into an array of pointer to size_t.
Should be an array of size_t

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
+			else if (guess[i] == number[0] || guess[i] == number[1] || guess[i] == number[2])
 			{
-				printf("Out! ");
+				outCount++;
```
Read the rule again.

ExtremeSkepticist replied:
Yes right. But, in the if statement, you are comparing if guess[i] matches with any values of indexes of number, and you increment outCount++. It should be ballCount++

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -2,15 +2,25 @@
+
+void Init(guess* player, guess* computer)
+{
+
```
Delete this new line

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
+		size_t outCount = 0;
+		size_t ballCount = 0;
+		size_t* guess[] = {player->guess1, player->guess2, player->guess3};
+		size_t* number[] = {computer->guess1, computer->guess2, computer->guess3, computer->guess1, computer->guess2};
```
Erase computer->guess1 and computer->guess 2 after computer->guess3

ExtremeSkepticist replied:
Same as the comment above. Should be an array of size_t.

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -23,81 +33,85 @@ void Init(int* number1, int* number2, int* number3)
 
-		if ((numbersEntered = sscanf_s(line, "%d %d %d",
-			guess1, guess2, guess3)) == 3)
+		if ((sscanf_s(line, "%u %u %u", guess1, guess2, guess3)) == 3)
```
You need to make sure if all numbers entered are distinct.

Kortigus committed:
-Fixed array of size_t pointers that should have been array of size_t
-Removed uneccesary variables
-Now checks if each value given is distinct when user is choosing numbers

ExtremeSkepticist commented on NumberBaseball/NumberBaseball.c:
```diff
@@ -59,11 +45,12 @@ void Update(guess* player, int* entry)
-			player->guess3 = test3;
+			if (player->guess1 == player->guess2 || player->guess1 == player->guess3 || player->guess2 == player->guess3)
+			{
+				return *entry = -1;
```
*entry = -1;
return;

Kortigus committed:
-Fixed error (returning a value assigned to pointer)
